### PR TITLE
use look behind in regex

### LINKED ISSUE #3: Use lookbehinds to improve the regex
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/issue-regex/issues/3)

[Lookbehinds are available in Node.js 10](http://node.green/#ES2018-features--RegExp-Lookbehind-Assertions), so we can either wait until we target that or do a conditional.

Would fix https://github.com/sindresorhus/issue-regex/blob/7958ee360b723207c279b82f5320941cac3cb096/test.js#L68

> [elyager](https://issuehunt.io/u/elyager) earned $20.00 by resolving this issue!
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/issue-regex/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by yaodingyd**
Fix #3 
Only target node 10

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -67,7 +67,6 @@ Should NOT match:
 		'#123',
 		'#123',
 		'another/repo#123',
-		'not/repo#123', // Current limitation; this match should not be here
```
Comment (kevva) [inline test.js:70]: Can you add `this/is/not/repo#123` to this line to make sure it isn't matched?

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,3 @@
 language: node_js
 node_js:
   - '10'
-  - '8'
```
Comment (kevva) [inline .travis.yml:4]: You will also have to set `node` to `>=10` in `engines` in `package.json`.

**Review by kevva — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (yaodingyd) [reply .travis.yml]: I don't understand what you mean. I thought this should be removed from expected

**Comment by sindresorhus** [comment]
I decided to go with #5 as it has a better regex and more complete tests.

File: index.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
 'use strict';
 
-// https://regex101.com/r/SQrOlx/12
-module.exports = () => /(?:\w[\w-.]+\/\w[\w-.]+|\B)#[1-9]\d*\b/g;
+// https://regex101.com/r/SQrOlx/13
+module.exports = () => /(?:(?<=[\s<[(])\w[\w-.]+\/\w[\w-.]+|\B)#[1-9]\d*\b/g;
```
Comment (sindresorhus) [inline index.js:4]: `\s<[(` is too specific to the fixture. Yes, it works, but there are many other real-world scenarios it misses by being so strict. The regex in #5 instead just says that anything is allowed before the issue except for a valid issue character.

**Pull request closed without merging**